winpr/libwinpr/sspi/test/TestInitializeSecurityContext.c

```diff
@@ -40,10 +40,19 @@ int TestInitializeSecurityContext(int argc, char* argv[])
 	cbMaxLen = pPackageInfo->cbMaxToken;
 
 	identity.User = (UINT16*) _strdup(test_User);
-	identity.UserLength = sizeof(test_User);
 	identity.Domain = (UINT16*) _strdup(test_Domain);
-	identity.DomainLength = sizeof(test_Domain);
 	identity.Password = (UINT16*) _strdup(test_Password);
+	if (!identity.User || !identity.Domain || !identity.Password)
+	{
+		free(identity.User);
+		free(identity.Domain);
+		free(identity.Password);
+		fprintf(stderr, "Memory allocation failed\n");
+		return -1;
+	}
+
+	identity.UserLength = sizeof(test_User);
+	identity.DomainLength = sizeof(test_Domain);
 	identity.PasswordLength = sizeof(test_Password);
```

Comment:
Not your code but like above these two sizeof() calls are wrong since it gives the pointer size !!